tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ImdsV2Tests.cs

```diff
@@ -82,8 +94,20 @@ private async Task<IManagedIdentityApplication> CreateManagedIdentityAsync(
                 .WithRetryPolicyFactory(_testRetryPolicyFactory)
                 .WithCsrFactory(_testCsrFactory);
 
-            
-            
+            // Choose deterministic key source for tests.
+            if (useKeyGuardKeys)
+            {
+                // Force KeyGuard keys to deterministically exercise the attestation path.
+                miBuilder.WithManagedIdentityKeyProviderForTests(new TestKeyGuardManagedIdentityKeyProvider());
+            }
+            else if (useInMemoryKeys)
+            {
+                // Default for bearer tests: no attestation.
+                miBuilder.WithManagedIdentityKeyProviderForTests(new InMemoryManagedIdentityKeyProvider());
+            }
+
+            // Disabling shared cache options to avoid cross test pollution.
+            miBuilder.Config.AccessorOptions = null;
```

Comment:
I don't think we need this anymore. Didn't Bogdan's PR for ResetCacheForTests remove the need for this?